pyScienceMode2/rehastim_interface.py

```diff
@@ -371,3 +367,527 @@ def get_motomed_torque(self) -> float:
+        return stim_status, voltage_level
+
+    def get_battery_status(self) -> tuple:
+        """
+        Get the battery status (battery level and battery voltage). General Level command.
+
+        Returns
+        -------
+        tuple
+        battery_level : int
+            Battery level.
+        battery_voltage : float
+            Battery voltage.
+        """
+        battery_status_ack = sciencemode.ffi.new("Smpt_get_battery_status_ack*")
+        packet_number = self.get_next_packet_number()
+        sciencemode.lib.smpt_send_get_battery_status(self.device, packet_number)
+
+        if self.show_log is True:
+            print("Command sent to rehastim:", self.TypeRehap24(sciencemode.lib.Smpt_Cmd_Get_Battery_Status).name)
+
+        self._get_last_ack()
+        ret = sciencemode.lib.smpt_get_get_battery_status_ack(self.device, battery_status_ack)
+        battery_level = f"battery level : {battery_status_ack.battery_level}"
+        battery_voltage = f"battery voltage : {battery_status_ack.battery_voltage}"
+        return battery_level, battery_voltage
+
+    def get_main_status(self):
+        """
+        Get the main status. General Level command.
+
+        Returns
+        -------
+        main_status : int
+            Main status.
+        """
+        main_status_ack = sciencemode.ffi.new("Smpt_get_main_status_ack*")
+        packet_number = self.get_next_packet_number()
+        sciencemode.lib.smpt_send_get_main_status(self.device, packet_number)
+
+        if self.show_log is True:
+            print("Command sent to rehastim:", self.TypeRehap24(sciencemode.lib.Smpt_Cmd_Get_Main_Status).name)
+
+        self._get_last_ack()
+        ret = sciencemode.lib.smpt_get_get_main_status_ack(self.device, main_status_ack)
+        main_status = f"main status : {main_status_ack.main_status}"
+        return main_status
+
+    def reset(self):
+        """
+        Reset the device. General Level command.
+        """
+        packet_number = self.get_next_packet_number()
+        ret = sciencemode.lib.smpt_send_reset(self.device, packet_number)
+
+        if self.show_log is True:
+            print("Command sent to rehastim:", self.TypeRehap24(sciencemode.lib.Smpt_Cmd_Reset).name)
+        self._get_last_ack()
+
+    def get_all(self):
+        """
+        Get all the device information. General Level command.
+        """
+        extended_version_success = self.get_extended_version()
+        device_id_success = self.get_device_id()
+        stim_status_success = self.get_stim_status()
+        battery_status_success = self.get_battery_status()
+        main_status_success = self.get_main_status()
+
+        return (
+            extended_version_success,
+            device_id_success,
+            stim_status_success,
+            battery_status_success,
+            main_status_success,
+        )
+
+    @staticmethod
+    def channel_number_to_channel_connector(no_channel):
```

Comment:
If static method put it elsewhere (like in the utils file) of if it stay here make it private.

Reply:
i've made it private because if I put it in the utils file, i must import the sciencemodep24 library again in this file